CMakeLists.txt

```diff
@@ -187,14 +187,10 @@ if(WIN32 AND ONEMKL_SYCL_IMPLEMENTATION STREQUAL "dpc++")
   endforeach()
   set(CMAKE_CXX_COMPILE_OBJECT "<CMAKE_CXX_COMPILER> -fsycl /nologo <DEFINES> <INCLUDES> /EHsc <FLAGS> /Fo<OBJECT> -c <SOURCE>")
   set(CMAKE_CXX_CREATE_STATIC_LIBRARY "lib /nologo <OBJECTS> /out:<TARGET>")
-  set(CMAKE_CXX_LINK_EXECUTABLE "<CMAKE_CXX_COMPILER> -fsycl -fsycl-device-code-split=per_kernel /nologo <OBJECTS> -o <TARGET> <LINK_LIBRARIES>")
-  set(MKL_SYCL_LIB "")
-  if(ENABLE_MKLGPU_BACKEND OR ENABLE_MKLCPU_BACKEND)
-    find_library(MKL_SYCL_LIB NAMES mkl_sycl
-      HINTS $ENV{MKLROOT} ${MKL_ROOT}
-      PATH_SUFFIXES lib/intel64)
+  if(CMAKE_VERSION VERSION_LESS "3.25.2")
```

Comment:
What happens with a newer version of CMake? Do newer versions somehow automatically know how to link the executable and library?

Reply:
Yes, because the Intel C/C++ compiler's official support was added to CMake starting 3.25.2.

Reply:
That's fine for `-fsycl` etc I guess but how can it possibly know `<OBJECTS> -o <TARGET> <LINK_LIBRARIES>`? What am I missing?

Reply:
We had to redefine the link line completely because by default only VS linker is used, but in case of SYCL objects we need SYCL compiler. For this PR I also noticed that we still provide static Intel oneMKL libraires to the linker instead of compiler, so I've changed the order in order to align it with the recommended link line.

This is how the link line looks like with our custom link line:
```
icx.exe -fsycl -fsycl-device-code-split=per_kernel /nologo                              #fsycl options from our custom link line
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_level1.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_level2.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_level3.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_batch.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_extensions.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_wrappers.cpp.obj  
-fsycl  -fsycl-device-code-split=per_kernel                                             #fsycl options from MKLConfig.cmake
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_sycl.lib  
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_intel_ilp64.lib  
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_tbb_thread.lib  
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_core.lib  
tbb12.lib  sycl6.lib  OpenCL.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib
 /link /out:bin\onemkl_blas_mklgpu.dll /implib:lib\onemkl_blas_mklgpu.lib /pdb:bin\onemkl_blas_mklgpu.pdb /dll /version:0.0 && cd ."
   Creating library lib\onemkl_blas_mklgpu.lib and object lib\onemkl_blas_mklgpu.exp
   Creating library lib\onemkl_blas_mklgpu.lib and object lib\onemkl_blas_mklgpu.exp
```

And this is how the link line looks like in case of using cmake 3.25.2
```
C:\Users\mkraynyu\Downloads\my_cache\3.25.2\win\bin\cmake.exe
-E vs_link_dll --intdir=bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu.dir 
--rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe 
--mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  
-- icx.exe /nologo 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_level1.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_level2.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_level3.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_batch.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_extensions.cpp.obj 
bin\blas\backends\mklgpu\CMakeFiles\onemkl_blas_mklgpu_obj.dir\mklgpu_wrappers.cpp.obj  
-LD /Qoption,link,/machine:x64 /INCREMENTAL:NO  
-fsycl  -fsycl-device-code-split=per_kernel
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_sycl.lib
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_intel_ilp64.lib  
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_tbb_thread.lib  
C:\Users\mkraynyu\Downloads\__release_win\lib\mkl_core.lib  
tbb12.lib  sycl6.lib  OpenCL.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib
-link /out:bin\onemkl_blas_mklgpu.dll /implib:lib\onemkl_blas_mklgpu.lib /pdb:bin\onemkl_blas_mklgpu.pdb /version:0.0  && cd ."
```

Reply:
Got it, thanks.